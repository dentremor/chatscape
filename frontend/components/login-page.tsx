import { Button, Title, Tooltip } from "@patternfly/react-core";
import { DownloadIcon, GlobeIcon } from "@patternfly/react-icons";
import Image from "next/image";
import logo from "../public/logo-light.png";

export const LoginPage = ({
  installPWA,
  logIn,
}: {
  installPWA?: Function;
  logIn: () => void;
}) => (
  <div className="pf-x-login-page pf-u-h-100 pf-u-display-flex pf-u-align-items-center pf-u-justify-content-center pf-u-flex-direction-column pf-u-p-lg">
    <div className="pf-u-flex-1 pf-u-display-flex pf-u-align-items-center pf-u-justify-content-center pf-u-flex-direction-column pf-u-p-md">
      <Image src={logo} alt="ChatScape logo" className="pf-x-logo" priority />

      <Title headingLevel="h3" className="pf-u-mt-sm pf-u-mb-lg pf-x-subtitle">
        Scalable Serverless Chat
      </Title>

      <div className="pf-x-ctas">
        <Tooltip
          trigger={installPWA ? "manual" : undefined}
          isVisible={installPWA ? false : undefined}
          content={
            <div>
              Your browser doesn&apos;t support PWAs, please use Chrome, Edge or
              another compatible browser to install the app or add it to your
              homescreen manually.
            </div>
          }
        >
          <Button
            variant="primary"
            icon={<DownloadIcon />}
            className={
              "pf-u-mr-0 pf-u-mr-sm-on-sm pf-u-mb-sm pf-u-mb-0-on-sm " +
              (installPWA ? "" : "pf-m-unusable")
            }
            onClick={async () => installPWA?.()}
          >
            Install the app
          </Button>
        </Tooltip>{" "}
        <Button
          variant="link"
          className="pf-u-mb-sm pf-u-mb-0-on-sm"
          onClick={logIn}
        >
          Open in browser <GlobeIcon />
        </Button>
      </div>
    </div>

    <div className="pf-u-w-100 pf-u-pt">
      <div className="pf-l-flex pf-m-justify-content-space-between pf-x-footer">
        <div className="pf-l-flex__item">
          <a href="https://github.com/pojntfx/chatscape" target="_blank">
            Â© AGPL-3.0 2023 Felicitas Pojtinger
          </a>
        </div>
        <div className="pf-l-flex__item">
          <a href="https://felicitas.pojtinger.com/imprint" target="_blank">
            Imprint
          </a>
        </div>
      </div>
    </div>
  </div>
);
