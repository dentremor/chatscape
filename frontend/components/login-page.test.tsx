import { render } from "@testing-library/react";
import { LoginPage } from "./login-page";

describe("LoginPage", () => {
  it("renders with installPWA set", () => {
    expect(
      render(<LoginPage installPWA={jest.fn()} logIn={jest.fn()} />)
    ).toMatchSnapshot();
  });

  it("renders without installPWA set", () => {
    expect(
      render(<LoginPage installPWA={jest.fn()} logIn={jest.fn()} />)
    ).toMatchSnapshot();
  });
});
