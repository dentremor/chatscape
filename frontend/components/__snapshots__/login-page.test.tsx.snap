// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginPage renders with installPWA set 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="pf-x-login-page pf-u-h-100 pf-u-display-flex pf-u-align-items-center pf-u-justify-content-center pf-u-flex-direction-column pf-u-p-lg"
      >
        <div
          class="pf-u-flex-1 pf-u-display-flex pf-u-align-items-center pf-u-justify-content-center pf-u-flex-direction-column pf-u-p-md"
        >
          <img
            alt="ChatScape logo"
            class="pf-x-logo"
            data-nimg="1"
            decoding="async"
            fetchpriority="high"
            height="40"
            src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
            srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
            style="color: transparent;"
            width="40"
          />
          <h3
            class="pf-c-title pf-m-lg pf-u-mt-sm pf-u-mb-lg pf-x-subtitle"
            data-ouia-component-id="OUIA-Generated-Title-1"
            data-ouia-component-type="PF4/Title"
            data-ouia-safe="true"
          >
            Scalable Serverless Chat
          </h3>
          <div
            class="pf-x-ctas"
          >
            <button
              aria-disabled="false"
              class="pf-c-button pf-m-primary pf-u-mr-0 pf-u-mr-sm-on-sm pf-u-mb-sm pf-u-mb-0-on-sm "
              data-ouia-component-id="OUIA-Generated-Button-primary-1"
              data-ouia-component-type="PF4/Button"
              data-ouia-safe="true"
              type="button"
            >
              <span
                class="pf-c-button__icon pf-m-start"
              >
                <svg
                  aria-hidden="true"
                  fill="currentColor"
                  height="1em"
                  role="img"
                  style="vertical-align: -0.125em;"
                  viewBox="0 0 512 512"
                  width="1em"
                >
                  <path
                    d="M216 0h80c13.3 0 24 10.7 24 24v168h87.7c17.8 0 26.7 21.5 14.1 34.1L269.7 378.3c-7.5 7.5-19.8 7.5-27.3 0L90.1 226.1c-12.6-12.6-3.7-34.1 14.1-34.1H192V24c0-13.3 10.7-24 24-24zm296 376v112c0 13.3-10.7 24-24 24H24c-13.3 0-24-10.7-24-24V376c0-13.3 10.7-24 24-24h146.7l49 49c20.1 20.1 52.5 20.1 72.6 0l49-49H488c13.3 0 24 10.7 24 24zm-124 88c0-11-9-20-20-20s-20 9-20 20 9 20 20 20 20-9 20-20zm64 0c0-11-9-20-20-20s-20 9-20 20 9 20 20 20 20-9 20-20z"
                  />
                </svg>
              </span>
              Install the app
            </button>
             
            <button
              aria-disabled="false"
              class="pf-c-button pf-m-link pf-u-mb-sm pf-u-mb-0-on-sm"
              data-ouia-component-id="OUIA-Generated-Button-link-1"
              data-ouia-component-type="PF4/Button"
              data-ouia-safe="true"
              type="button"
            >
              Open in browser 
              <svg
                aria-hidden="true"
                fill="currentColor"
                height="1em"
                role="img"
                style="vertical-align: -0.125em;"
                viewBox="0 0 496 512"
                width="1em"
              >
                <path
                  d="M336.5 160C322 70.7 287.8 8 248 8s-74 62.7-88.5 152h177zM152 256c0 22.2 1.2 43.5 3.3 64h185.3c2.1-20.5 3.3-41.8 3.3-64s-1.2-43.5-3.3-64H155.3c-2.1 20.5-3.3 41.8-3.3 64zm324.7-96c-28.6-67.9-86.5-120.4-158-141.6 24.4 33.8 41.2 84.7 50 141.6h108zM177.2 18.4C105.8 39.6 47.8 92.1 19.3 160h108c8.7-56.9 25.5-107.8 49.9-141.6zM487.4 192H372.7c2.1 21 3.3 42.5 3.3 64s-1.2 43-3.3 64h114.6c5.5-20.5 8.6-41.8 8.6-64s-3.1-43.5-8.5-64zM120 256c0-21.5 1.2-43 3.3-64H8.6C3.2 212.5 0 233.8 0 256s3.2 43.5 8.6 64h114.6c-2-21-3.2-42.5-3.2-64zm39.5 96c14.5 89.3 48.7 152 88.5 152s74-62.7 88.5-152h-177zm159.3 141.6c71.4-21.2 129.4-73.7 158-141.6h-108c-8.8 56.9-25.6 107.8-50 141.6zM19.3 352c28.6 67.9 86.5 120.4 158 141.6-24.4-33.8-41.2-84.7-50-141.6h-108z"
                />
              </svg>
            </button>
          </div>
        </div>
        <div
          class="pf-u-w-100 pf-u-pt"
        >
          <div
            class="pf-l-flex pf-m-justify-content-space-between pf-x-footer"
          >
            <div
              class="pf-l-flex__item"
            >
              <a
                href="https://github.com/pojntfx/chatscape"
                target="_blank"
              >
                © AGPL-3.0 2023 Felicitas Pojtinger
              </a>
            </div>
            <div
              class="pf-l-flex__item"
            >
              <a
                href="https://felicitas.pojtinger.com/imprint"
                target="_blank"
              >
                Imprint
              </a>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="pf-x-login-page pf-u-h-100 pf-u-display-flex pf-u-align-items-center pf-u-justify-content-center pf-u-flex-direction-column pf-u-p-lg"
    >
      <div
        class="pf-u-flex-1 pf-u-display-flex pf-u-align-items-center pf-u-justify-content-center pf-u-flex-direction-column pf-u-p-md"
      >
        <img
          alt="ChatScape logo"
          class="pf-x-logo"
          data-nimg="1"
          decoding="async"
          fetchpriority="high"
          height="40"
          src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
          srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
          style="color: transparent;"
          width="40"
        />
        <h3
          class="pf-c-title pf-m-lg pf-u-mt-sm pf-u-mb-lg pf-x-subtitle"
          data-ouia-component-id="OUIA-Generated-Title-1"
          data-ouia-component-type="PF4/Title"
          data-ouia-safe="true"
        >
          Scalable Serverless Chat
        </h3>
        <div
          class="pf-x-ctas"
        >
          <button
            aria-disabled="false"
            class="pf-c-button pf-m-primary pf-u-mr-0 pf-u-mr-sm-on-sm pf-u-mb-sm pf-u-mb-0-on-sm "
            data-ouia-component-id="OUIA-Generated-Button-primary-1"
            data-ouia-component-type="PF4/Button"
            data-ouia-safe="true"
            type="button"
          >
            <span
              class="pf-c-button__icon pf-m-start"
            >
              <svg
                aria-hidden="true"
                fill="currentColor"
                height="1em"
                role="img"
                style="vertical-align: -0.125em;"
                viewBox="0 0 512 512"
                width="1em"
              >
                <path
                  d="M216 0h80c13.3 0 24 10.7 24 24v168h87.7c17.8 0 26.7 21.5 14.1 34.1L269.7 378.3c-7.5 7.5-19.8 7.5-27.3 0L90.1 226.1c-12.6-12.6-3.7-34.1 14.1-34.1H192V24c0-13.3 10.7-24 24-24zm296 376v112c0 13.3-10.7 24-24 24H24c-13.3 0-24-10.7-24-24V376c0-13.3 10.7-24 24-24h146.7l49 49c20.1 20.1 52.5 20.1 72.6 0l49-49H488c13.3 0 24 10.7 24 24zm-124 88c0-11-9-20-20-20s-20 9-20 20 9 20 20 20 20-9 20-20zm64 0c0-11-9-20-20-20s-20 9-20 20 9 20 20 20 20-9 20-20z"
                />
              </svg>
            </span>
            Install the app
          </button>
           
          <button
            aria-disabled="false"
            class="pf-c-button pf-m-link pf-u-mb-sm pf-u-mb-0-on-sm"
            data-ouia-component-id="OUIA-Generated-Button-link-1"
            data-ouia-component-type="PF4/Button"
            data-ouia-safe="true"
            type="button"
          >
            Open in browser 
            <svg
              aria-hidden="true"
              fill="currentColor"
              height="1em"
              role="img"
              style="vertical-align: -0.125em;"
              viewBox="0 0 496 512"
              width="1em"
            >
              <path
                d="M336.5 160C322 70.7 287.8 8 248 8s-74 62.7-88.5 152h177zM152 256c0 22.2 1.2 43.5 3.3 64h185.3c2.1-20.5 3.3-41.8 3.3-64s-1.2-43.5-3.3-64H155.3c-2.1 20.5-3.3 41.8-3.3 64zm324.7-96c-28.6-67.9-86.5-120.4-158-141.6 24.4 33.8 41.2 84.7 50 141.6h108zM177.2 18.4C105.8 39.6 47.8 92.1 19.3 160h108c8.7-56.9 25.5-107.8 49.9-141.6zM487.4 192H372.7c2.1 21 3.3 42.5 3.3 64s-1.2 43-3.3 64h114.6c5.5-20.5 8.6-41.8 8.6-64s-3.1-43.5-8.5-64zM120 256c0-21.5 1.2-43 3.3-64H8.6C3.2 212.5 0 233.8 0 256s3.2 43.5 8.6 64h114.6c-2-21-3.2-42.5-3.2-64zm39.5 96c14.5 89.3 48.7 152 88.5 152s74-62.7 88.5-152h-177zm159.3 141.6c71.4-21.2 129.4-73.7 158-141.6h-108c-8.8 56.9-25.6 107.8-50 141.6zM19.3 352c28.6 67.9 86.5 120.4 158 141.6-24.4-33.8-41.2-84.7-50-141.6h-108z"
              />
            </svg>
          </button>
        </div>
      </div>
      <div
        class="pf-u-w-100 pf-u-pt"
      >
        <div
          class="pf-l-flex pf-m-justify-content-space-between pf-x-footer"
        >
          <div
            class="pf-l-flex__item"
          >
            <a
              href="https://github.com/pojntfx/chatscape"
              target="_blank"
            >
              © AGPL-3.0 2023 Felicitas Pojtinger
            </a>
          </div>
          <div
            class="pf-l-flex__item"
          >
            <a
              href="https://felicitas.pojtinger.com/imprint"
              target="_blank"
            >
              Imprint
            </a>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`LoginPage renders without installPWA set 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="pf-x-login-page pf-u-h-100 pf-u-display-flex pf-u-align-items-center pf-u-justify-content-center pf-u-flex-direction-column pf-u-p-lg"
      >
        <div
          class="pf-u-flex-1 pf-u-display-flex pf-u-align-items-center pf-u-justify-content-center pf-u-flex-direction-column pf-u-p-md"
        >
          <img
            alt="ChatScape logo"
            class="pf-x-logo"
            data-nimg="1"
            decoding="async"
            fetchpriority="high"
            height="40"
            src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
            srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
            style="color: transparent;"
            width="40"
          />
          <h3
            class="pf-c-title pf-m-lg pf-u-mt-sm pf-u-mb-lg pf-x-subtitle"
            data-ouia-component-id="OUIA-Generated-Title-2"
            data-ouia-component-type="PF4/Title"
            data-ouia-safe="true"
          >
            Scalable Serverless Chat
          </h3>
          <div
            class="pf-x-ctas"
          >
            <button
              aria-disabled="false"
              class="pf-c-button pf-m-primary pf-u-mr-0 pf-u-mr-sm-on-sm pf-u-mb-sm pf-u-mb-0-on-sm "
              data-ouia-component-id="OUIA-Generated-Button-primary-2"
              data-ouia-component-type="PF4/Button"
              data-ouia-safe="true"
              type="button"
            >
              <span
                class="pf-c-button__icon pf-m-start"
              >
                <svg
                  aria-hidden="true"
                  fill="currentColor"
                  height="1em"
                  role="img"
                  style="vertical-align: -0.125em;"
                  viewBox="0 0 512 512"
                  width="1em"
                >
                  <path
                    d="M216 0h80c13.3 0 24 10.7 24 24v168h87.7c17.8 0 26.7 21.5 14.1 34.1L269.7 378.3c-7.5 7.5-19.8 7.5-27.3 0L90.1 226.1c-12.6-12.6-3.7-34.1 14.1-34.1H192V24c0-13.3 10.7-24 24-24zm296 376v112c0 13.3-10.7 24-24 24H24c-13.3 0-24-10.7-24-24V376c0-13.3 10.7-24 24-24h146.7l49 49c20.1 20.1 52.5 20.1 72.6 0l49-49H488c13.3 0 24 10.7 24 24zm-124 88c0-11-9-20-20-20s-20 9-20 20 9 20 20 20 20-9 20-20zm64 0c0-11-9-20-20-20s-20 9-20 20 9 20 20 20 20-9 20-20z"
                  />
                </svg>
              </span>
              Install the app
            </button>
             
            <button
              aria-disabled="false"
              class="pf-c-button pf-m-link pf-u-mb-sm pf-u-mb-0-on-sm"
              data-ouia-component-id="OUIA-Generated-Button-link-2"
              data-ouia-component-type="PF4/Button"
              data-ouia-safe="true"
              type="button"
            >
              Open in browser 
              <svg
                aria-hidden="true"
                fill="currentColor"
                height="1em"
                role="img"
                style="vertical-align: -0.125em;"
                viewBox="0 0 496 512"
                width="1em"
              >
                <path
                  d="M336.5 160C322 70.7 287.8 8 248 8s-74 62.7-88.5 152h177zM152 256c0 22.2 1.2 43.5 3.3 64h185.3c2.1-20.5 3.3-41.8 3.3-64s-1.2-43.5-3.3-64H155.3c-2.1 20.5-3.3 41.8-3.3 64zm324.7-96c-28.6-67.9-86.5-120.4-158-141.6 24.4 33.8 41.2 84.7 50 141.6h108zM177.2 18.4C105.8 39.6 47.8 92.1 19.3 160h108c8.7-56.9 25.5-107.8 49.9-141.6zM487.4 192H372.7c2.1 21 3.3 42.5 3.3 64s-1.2 43-3.3 64h114.6c5.5-20.5 8.6-41.8 8.6-64s-3.1-43.5-8.5-64zM120 256c0-21.5 1.2-43 3.3-64H8.6C3.2 212.5 0 233.8 0 256s3.2 43.5 8.6 64h114.6c-2-21-3.2-42.5-3.2-64zm39.5 96c14.5 89.3 48.7 152 88.5 152s74-62.7 88.5-152h-177zm159.3 141.6c71.4-21.2 129.4-73.7 158-141.6h-108c-8.8 56.9-25.6 107.8-50 141.6zM19.3 352c28.6 67.9 86.5 120.4 158 141.6-24.4-33.8-41.2-84.7-50-141.6h-108z"
                />
              </svg>
            </button>
          </div>
        </div>
        <div
          class="pf-u-w-100 pf-u-pt"
        >
          <div
            class="pf-l-flex pf-m-justify-content-space-between pf-x-footer"
          >
            <div
              class="pf-l-flex__item"
            >
              <a
                href="https://github.com/pojntfx/chatscape"
                target="_blank"
              >
                © AGPL-3.0 2023 Felicitas Pojtinger
              </a>
            </div>
            <div
              class="pf-l-flex__item"
            >
              <a
                href="https://felicitas.pojtinger.com/imprint"
                target="_blank"
              >
                Imprint
              </a>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="pf-x-login-page pf-u-h-100 pf-u-display-flex pf-u-align-items-center pf-u-justify-content-center pf-u-flex-direction-column pf-u-p-lg"
    >
      <div
        class="pf-u-flex-1 pf-u-display-flex pf-u-align-items-center pf-u-justify-content-center pf-u-flex-direction-column pf-u-p-md"
      >
        <img
          alt="ChatScape logo"
          class="pf-x-logo"
          data-nimg="1"
          decoding="async"
          fetchpriority="high"
          height="40"
          src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
          srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
          style="color: transparent;"
          width="40"
        />
        <h3
          class="pf-c-title pf-m-lg pf-u-mt-sm pf-u-mb-lg pf-x-subtitle"
          data-ouia-component-id="OUIA-Generated-Title-2"
          data-ouia-component-type="PF4/Title"
          data-ouia-safe="true"
        >
          Scalable Serverless Chat
        </h3>
        <div
          class="pf-x-ctas"
        >
          <button
            aria-disabled="false"
            class="pf-c-button pf-m-primary pf-u-mr-0 pf-u-mr-sm-on-sm pf-u-mb-sm pf-u-mb-0-on-sm "
            data-ouia-component-id="OUIA-Generated-Button-primary-2"
            data-ouia-component-type="PF4/Button"
            data-ouia-safe="true"
            type="button"
          >
            <span
              class="pf-c-button__icon pf-m-start"
            >
              <svg
                aria-hidden="true"
                fill="currentColor"
                height="1em"
                role="img"
                style="vertical-align: -0.125em;"
                viewBox="0 0 512 512"
                width="1em"
              >
                <path
                  d="M216 0h80c13.3 0 24 10.7 24 24v168h87.7c17.8 0 26.7 21.5 14.1 34.1L269.7 378.3c-7.5 7.5-19.8 7.5-27.3 0L90.1 226.1c-12.6-12.6-3.7-34.1 14.1-34.1H192V24c0-13.3 10.7-24 24-24zm296 376v112c0 13.3-10.7 24-24 24H24c-13.3 0-24-10.7-24-24V376c0-13.3 10.7-24 24-24h146.7l49 49c20.1 20.1 52.5 20.1 72.6 0l49-49H488c13.3 0 24 10.7 24 24zm-124 88c0-11-9-20-20-20s-20 9-20 20 9 20 20 20 20-9 20-20zm64 0c0-11-9-20-20-20s-20 9-20 20 9 20 20 20 20-9 20-20z"
                />
              </svg>
            </span>
            Install the app
          </button>
           
          <button
            aria-disabled="false"
            class="pf-c-button pf-m-link pf-u-mb-sm pf-u-mb-0-on-sm"
            data-ouia-component-id="OUIA-Generated-Button-link-2"
            data-ouia-component-type="PF4/Button"
            data-ouia-safe="true"
            type="button"
          >
            Open in browser 
            <svg
              aria-hidden="true"
              fill="currentColor"
              height="1em"
              role="img"
              style="vertical-align: -0.125em;"
              viewBox="0 0 496 512"
              width="1em"
            >
              <path
                d="M336.5 160C322 70.7 287.8 8 248 8s-74 62.7-88.5 152h177zM152 256c0 22.2 1.2 43.5 3.3 64h185.3c2.1-20.5 3.3-41.8 3.3-64s-1.2-43.5-3.3-64H155.3c-2.1 20.5-3.3 41.8-3.3 64zm324.7-96c-28.6-67.9-86.5-120.4-158-141.6 24.4 33.8 41.2 84.7 50 141.6h108zM177.2 18.4C105.8 39.6 47.8 92.1 19.3 160h108c8.7-56.9 25.5-107.8 49.9-141.6zM487.4 192H372.7c2.1 21 3.3 42.5 3.3 64s-1.2 43-3.3 64h114.6c5.5-20.5 8.6-41.8 8.6-64s-3.1-43.5-8.5-64zM120 256c0-21.5 1.2-43 3.3-64H8.6C3.2 212.5 0 233.8 0 256s3.2 43.5 8.6 64h114.6c-2-21-3.2-42.5-3.2-64zm39.5 96c14.5 89.3 48.7 152 88.5 152s74-62.7 88.5-152h-177zm159.3 141.6c71.4-21.2 129.4-73.7 158-141.6h-108c-8.8 56.9-25.6 107.8-50 141.6zM19.3 352c28.6 67.9 86.5 120.4 158 141.6-24.4-33.8-41.2-84.7-50-141.6h-108z"
              />
            </svg>
          </button>
        </div>
      </div>
      <div
        class="pf-u-w-100 pf-u-pt"
      >
        <div
          class="pf-l-flex pf-m-justify-content-space-between pf-x-footer"
        >
          <div
            class="pf-l-flex__item"
          >
            <a
              href="https://github.com/pojntfx/chatscape"
              target="_blank"
            >
              © AGPL-3.0 2023 Felicitas Pojtinger
            </a>
          </div>
          <div
            class="pf-l-flex__item"
          >
            <a
              href="https://felicitas.pojtinger.com/imprint"
              target="_blank"
            >
              Imprint
            </a>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
